### PR TITLE
Unit tests for Module Repository

### PR DESCRIPTION
**Pull request by FrenjaminBanklin**
Added unit tests for everything relevant in the repository - copied from the private collections branch and rearranged/removed tests as needed to cover current code, minus all the other new additions from that branch. Modified existing code as needed to make it more testable, standardized filenames to '.jsx' for files with JSX and '.js' for files without.

Since this was copied from another branch with additional changes that were also fully tested, there may be some leftover code for the additional features that I missed when I was cleaning up.

### CODE REVIEW

**Review by zachberry — DISMISSED** [review]
Nice work!

File: packages/app/obojobo-repository/server/models/draft_summary.test.js [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+		return DraftSummary.fetchById('mockDraftId').catch(err => {
+			expect(logger.error).toHaveBeenCalledWith('fetchById Error', 'not found in db')
+			expect(err).toBe('Error Loading DraftSummary by id')
+		})
+	})
+
+	test('fetchByUserId generates the correct query and returns a DraftSummary object', () => {
+		db.any = jest.fn()
+		db.any.mockResolvedValueOnce(mockRawDraftSummary)
+
```
Comment (iturgeon) [inline packages/app/obojobo-repository/server/models/draft_summary.test.js:123]:
```suggestion
		expect.hasAssertions()
```

File: packages/app/obojobo-repository/server/models/draft_summary.test.js [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+				ON drafts_content.user_id = users.id
+			WHERE
+				drafts_content.draft_id = $[draftId]
+			ORDER BY
+				drafts_content.created_at DESC
+			LIMIT $[count];
+		`
+
+		return DraftSummary.fetchAllDraftRevisions('mockDraftId').then(history => {
+			expect(db.any).toHaveBeenCalledWith(query, {
```
Comment (iturgeon) [inline packages/app/obojobo-repository/server/models/draft_summary.test.js:200]: Quick idea, not sure, this maybe could use an 'expect string contains' against the important bits of the query?

File: packages/app/obojobo-repository/server/models/drafts_metadata.test.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	test('saveOrCreate returns a DraftsMetadata object', () => {
+		db.none = jest.fn()
+		db.none.mockResolvedValueOnce(mockRawDraftsMetadata)
+
+		const draftMetadata = new DraftsMetadata(mockRawDraftsMetadata)
+
+		expect.hasAssertions()
+
+		// Trying to match whitespace with the query that's actually running
+		const query = `
```
Comment (iturgeon) [inline packages/app/obojobo-repository/server/models/drafts_metadata.test.js:46]: another idea for these (if string contains isn't any good), use a snapshot test.

File: packages/app/obojobo-repository/server/services/search.test.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		const mockResponse = [{ ...mockRawUser }, { ...mockRawUser, id: 2 }, { ...mockRawUser, id: 3 }]
+
+		db.manyOrNone.mockResolvedValueOnce(mockResponse)
+
+		db.taskIf = jest.fn()
+		db.taskIf.mockImplementation(cb => cb(db))
+
+		expect.hasAssertions()
+
+		const manyOrNoneQueryString = `SELECT
```
Comment (iturgeon) [inline packages/app/obojobo-repository/server/services/search.test.js:38]: same notes on testing a match of this query string as above

File: packages/app/obojobo-repository/setupTests.js [hunk 5]
```diff
@@ -0,0 +1 @@
+require('obojobo-lib-utils/test-setup-chunks') // setup enzyme
```
Comment (iturgeon) [inline packages/app/obojobo-repository/setupTests.js:1]: just making sure this is needed (if so maybe we should rename test-setup-chunks)

File: packages/app/obojobo-repository/shared/actions/dashboard-actions.js [hunk 6]
```diff
@@ -229,7 +229,7 @@ const promptUserForModuleFileUpload = async () => {
 
 const moduleUploadFileSelected = (boundResolve, boundReject, event) => {
 	const file = event.target.files[0]
-	if (!file) boundResolve()
+	if (!file) return boundResolve()
```
Comment (iturgeon) [inline packages/app/obojobo-repository/shared/actions/dashboard-actions.js:232]: whoopsie.  Yay tests!

**Review by iturgeon — APPROVED** [review]
Looks pretty awesome to me

File: packages/app/obojobo-repository/server/models/collection.test.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+			id: 'mockCollectionId',
+			title: 'mockCollectionTitle',
+			user_id: userId,
+			created_at: new Date().toISOString()
+		}
+
+		db.one.mockResolvedValueOnce(mockNewRawCollection)
+
+		return CollectionModel.create('', userId).then(model => {
+			expect(model).toBeInstanceOf(CollectionModel)
```
[the remark this answers is not in the document]
Comment (FrenjaminBanklin) [reply packages/app/obojobo-repository/server/models/collection.test.js:87]: Ah, leftovers - when I made changes to the `RepositoryCollection` model in the private collections branch I also added logging for creation collection - that part got removed, I just forgot to rename the test. Currently, there's no logging on creation of a collection.

Writing a separate test to make sure `db.one()` is being called correctly is a fair point, though I think the `expect`s here match the intention of making sure that calling `.create` with no title will still return a collection model with the correct defaults.

File: packages/app/obojobo-repository/server/models/draft_summary.test.js [hunk 8]
```diff
@@ -0,0 +1,516 @@
...
+		WHERE drafts.deleted = FALSE
+		AND ${whereSQL}
+		WINDOW wnd AS (
+			PARTITION BY drafts_content.draft_id ORDER BY drafts_content.created_at
+			ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
+		)
+		ORDER BY updated_at DESC
+	`
+
+	const checkAgainstMockRawSummary = summary => {
```
[the remark this answers is not in the document]
Comment (FrenjaminBanklin) [reply packages/app/obojobo-repository/server/models/draft_summary.test.js:83]: Good point. Easy change.

File: packages/app/obojobo-repository/server/models/draft_summary.test.js (hunk 1 above)
Comment (FrenjaminBanklin) [reply packages/app/obojobo-repository/server/models/draft_summary.test.js:123]: Oops. Added.

File: packages/app/obojobo-repository/server/models/draft_summary.test.js (hunk 2 above)
Comment (FrenjaminBanklin) [reply packages/app/obojobo-repository/server/models/draft_summary.test.js:200]: I think I might have tried that, but it ended up feeling a bit kludgy for queries where there was a lot to look for. I had kind of hoped there was some jest magic to ignore initial indentation, but that didn't pan out either.

File: packages/app/obojobo-repository/server/models/drafts_metadata.test.js (hunk 3 above)
Comment (FrenjaminBanklin) [reply packages/app/obojobo-repository/server/models/drafts_metadata.test.js:46]: I'm mostly writing out queries like this to pass them into a `toHaveBeenCalledWith` to make sure we're sending the right queries to our `db` calls - maybe it's a feature of snapshots I'm not familiar with, but I'm not sure if they'd be applicable here.

File: packages/app/obojobo-repository/setupTests.js (hunk 5 above)
Comment (FrenjaminBanklin) [reply packages/app/obojobo-repository/setupTests.js:1]: Not sure about the name, but some of the tests use Enzyme so this'll definitely be necessary.

**Comment by FrenjaminBanklin** [comment]
There's still a weird Async timeout error coming from the `repository library route › get / returns the expected response` test that I haven't been able to figure out. If run in isolation (or even just alongside the other repository tests) it works fine, but when all of the Obojobo tests run it will sporadically time out - usually only just the first time the tests are run, and then a subsequent run will succeed without a hitch.

Another weird thing - something about this test failing bleeds into the next (completely unrelated) test, causing failures in that as well - although Jest will report the failure as coming from the first test. It's all very strange.

There's got to be some way of preventing this altogether, but since I can't even figure out how to reproduce it reliably I'm not sure how to approach it.

File: packages/app/obojobo-repository/server/models/drafts_metadata.test.js (hunk 3 above)
Comment (iturgeon) [reply packages/app/obojobo-repository/server/models/drafts_metadata.test.js:46]: Don't forget you can just extract a single argument to a function `db.one.calls[0][0]` (first argument of first call).

Another option is `.toHaveBeenCalledWith(expect. stringContaining('query'), {})`

perhaps `expect(db.one.calls[0][0].replace(/[\W\t\n\r]+/g, ' ')).toBe(expectedQuery.replace(/[\W\t\n\r]+/g, ' '))`

Comment (FrenjaminBanklin) [reply packages/app/obojobo-repository/server/models/drafts_metadata.test.js:46]: That could work. Might want to put it behind reusable function to denastify it for multiple uses, though. Probably a bit late to retrofit the tests now, but perhaps for a refactor later and/or all future tests doing string comparisons like this.

**Review by iturgeon — APPROVED** [review]

**Pull request merged**